crates/storage_impl/src/lib.rs

```diff
@@ -329,6 +329,21 @@ impl UniqueConstraints for diesel_models::Address {
     }
 }
 
+#[cfg(all(feature = "v2", feature = "payment_v2"))]
+impl UniqueConstraints for diesel_models::PaymentIntent {
+    fn unique_constraints(&self) -> Vec<String> {
+        vec![format!(
+            "pi_{}_{}",
+            self.merchant_id.get_string_repr(),
+            self.merchant_reference_id
+        )]
```

Comment:
This is not required add a note to remove this, id itself will give us unique_constraints